java-spiffe-core/src/main/java/io/spiffe/workloadapi/JwtSource.java

```diff
@@ -0,0 +1,256 @@
+     * Creates a new JWT source. It blocks until the initial update
+     * has been received from the Workload API or until the timeout configured
+     * through the system property `spiffe.newJwtSource.timeout` expires.
+     * If no timeout is configured, it blocks until it gets a JWT update from the Workload API.
+     * <p>
+     * It uses the default address socket endpoint from the environment variable to get the Workload API address.
+     *
+     * @return an instance of {@link JwtSource}, with the JWT bundles initialized
+     * @throws SocketEndpointAddressException if the address to the Workload API is not valid
+     * @throws JwtSourceException            if the source could not be initialized
+     */
+    public static JwtSource newSource() throws JwtSourceException, SocketEndpointAddressException {
+        JwtSourceOptions options = JwtSourceOptions.builder().build();
+        return newSource(options, DEFAULT_TIMEOUT);
+    }
+
+    /**
+     * Creates a new JWT source. It blocks until the initial update
+     * has been received from the Workload API or until the timeout configured
+     * through the system property `spiffe.newJwtSource.timeout` expires.
+     * If no timeout is configured, it blocks until it gets a JWT update from the Workload API.
+     * <p>
+     * It uses the default address socket endpoint from the environment variable to get the Workload API address.
+     *
+     * @param timeout Time to wait for the JWT bundles update. If the timeout is Zero, it will wait indefinitely.
+     * @return an instance of {@link JwtSource}, with the JWT bundles initialized
+     * @throws SocketEndpointAddressException if the address to the Workload API is not valid
+     * @throws JwtSourceException            if the source could not be initialized
+     */
+    public static JwtSource newSource(@NonNull final Duration timeout) throws JwtSourceException, SocketEndpointAddressException {
+        JwtSourceOptions options = JwtSourceOptions.builder().build();
+        return newSource(options, timeout);
+    }
+
+    /**
+     * Creates a new JWT source. It blocks until the initial update
+     * has been received from the Workload API or until the timeout configured
+     * through the system property `spiffe.newJwtSource.timeout` expires.
+     * If no timeout is configured, it blocks until it gets a JWT update from the Workload API.
+     * <p>
+     * It uses the default address socket endpoint from the environment variable to get the Workload API address.
+     *
+     * @param options {@link JwtSourceOptions}
+     * @return an instance of {@link JwtSource}, with the JWT bundles initialized
+     * @throws SocketEndpointAddressException if the address to the Workload API is not valid
+     * @throws JwtSourceException            if the source could not be initialized
+     */
+    public static JwtSource newSource(@NonNull final JwtSourceOptions options) throws JwtSourceException, SocketEndpointAddressException {
```

Comment:
I think it would be better to move the `timeout` option to the `JwtSourceOptions` class, because it can be considered a config of the source, and that class has a builder; this way we would have only 2 methods. What do you think ?

Reply:
That seems reasonable to me.

Reply:
Refactored both `JwtSource` and `X509Source` new methods. Now we have one method taking no parameters and using the defaults and another that takes an instances with the options.